src/backend/distributed/commands/utility_hook.c

```diff
@@ -94,6 +95,28 @@
 #define MARK_OBJECT_DISTRIBUTED \
 	"SELECT citus_internal.mark_object_distributed(%d, %s, %d, %s)"
 
+/*
+ * TwoPcStatementInfo is used to determine whether a statement is supported in 2PC
+ * and whether it should be marked as distributed in 2PC.
+ */
+typedef struct TwoPcStatementInfo
+{
+	int statementType;
+	bool markAsDistributed;
+} TwoPcStatementInfo;
```

Comment:
Maybe NonMainDbDistributedStatementInfo?

```suggestion
/*
 * NonMainDbDistributedStatementInfo is used to determine whether a statement is
 * supported from non-main databases and whether it should be marked as
 * distributed explicitly.
 */
typedef struct NonMainDbDistributedStatementInfo
{
	int statementType;
	bool explicitlyMarkAsDistributed;
} NonMainDbDistributedStatementInfo;
```

Reply:
Because "2PC statement info" doesn't really make sense because it's a whole different concept than supporting commands from a non-main database. "2PC logic" is simply where we hook into to support those features.